app/api/protocol/vaults/multicall/fetch-assets-via-multicall.ts

```diff
@@ -0,0 +1,125 @@
+import { JsonRpcProvider } from 'ethers'
+import _ from 'lodash'
+import type { Price, Token } from '../../gql/graphql'
+import { Multicall } from '@/shared'
+import type { MulticallRequest, MulticallResponse } from '@/shared'
+import { type ChainId, getRPCEndpoint } from '@/providers/wallet/wrappers/helpers'
+import ERC20ABI from '@/shared/web3/abi/ERC20.json'
+import { getCurrentPriceOfTokens } from '@/api/coin-gecko/price/useCurrentPriceOfTokens'
+import type { TokenSymbol } from '@/types'
+import { TokenOrder } from '@/types'
+
+interface IntermediateAssetModel {
+  name: string
+  symbol: string
+  address: string
+  decimals: number
+}
+
+/**
+ * Performs multicall request to fetch data for the specified ERC20 assets.
+ * @param chainId Current chain id.
+ * @param multicallAddress Multicall contract for the specified chain.
+ * @param assetAddresses List of ERC20 addresses.
+ * @returns List of enriched token model.
+ */
+export async function fetchAssetsViaMulticall(
+  chainId: ChainId,
+  multicallAddress: string,
+  assetAddresses: string[],
+): Promise<Token[]> {
+  const requests = assetAddresses.flatMap(createAssetRequests)
+  if (requests.length === 0)
+    return []
+
+  const provider = new JsonRpcProvider(getRPCEndpoint(chainId))
+  const multicall = new Multicall(multicallAddress, provider, requests)
+  const responses = await multicall.makeRequest()
+
+  // Convert raw multicall responses to the intermediate asset model
+  const intermediateAssetModels = _.chain(responses)
+    .chunk(createAssetRequests('-').length)
+    .map((responses, index) => mapResponseToIntermediateAssetModel(assetAddresses[index], responses))
+    .compact()
+    .value()
```

Comment:
Why everywhere lodash? Looks like code from 90x, it really hard to follow. Why not write it in regular way? You anyway hardcoded everywhere that it three specific requests per asset
```suggestion

 const data = response.map(({isSuccess, data}) => isSuccess ? data : null)
 // this part I would move to another function 
 const  intermediateAssetModels = []
 for(let i = 0; i += 3; i < requests.length) {
  const  name = data[i]
  const  symbol = data[i + 1]
  const  decimals = data[i + 2]
  const address = assetAddresses[i / 3]
  
  if(![name, symbol, decimals].includes(v => v === null))
   intermediateAssetModels.push({
      name: String(name),
      symbol: String(symbol),
      decimals: Number(decimals),
      address,
  })
 }
```

Reply:
discussed in private, decided to leave it for now